code/game/machinery/virtual_reality/vr_console.dm

```diff
@@ -1,227 +1,260 @@
+	var/datum/effect/effect/system/smoke_spread/bad/smoke
+
+	var/eject_dead = TRUE
+
+	use_power = 1
+	idle_power_usage = 15
+	active_power_usage = 200
+	light_color = "#FF0000"
+
+/obj/machinery/vr_sleeper/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/stack/material/glass/reinforced(src, 2)
+
+	RefreshParts()
+
+/obj/machinery/vr_sleeper/Initialize()
+	. = ..()
+	smoke = new
+	update_icon()
+
+/obj/machinery/vr_sleeper/Destroy()
+	. = ..()
+	go_out()
+
+/obj/machinery/vr_sleeper/process()
+	if(stat & (NOPOWER|BROKEN))
+		if(occupant)
+			go_out()
+			visible_message("<span class='notice'>\The [src] emits a low droning sound, before the pod door clicks open.</span>")
+		return
+	else if(eject_dead && occupant && occupant.stat == DEAD) // If someone dies somehow while inside, spit them out.
+		visible_message("<span class='warning'>\The [src] sounds an alarm, swinging its hatch open.</span>")
+		go_out()
+
+/obj/machinery/vr_sleeper/update_icon()
+	icon_state = "[base_state][occupant ? "1" : "0"]"
+
+/obj/machinery/vr_sleeper/Topic(href, href_list)
+	if(..())
+		return 1
+
+	if(usr == occupant)
+		to_chat(usr, "<span class='warning'>You can't reach the controls from the inside.</span>")
+		return
+
+	add_fingerprint(usr)
+
+	if(href_list["eject"])
+		go_out()
+
+	return 1
+
+/obj/machinery/vr_sleeper/attackby(var/obj/item/I, var/mob/user)
+	add_fingerprint(user)
+
+	if(occupant && (istype(I, /obj/item/device/healthanalyzer) || istype(I, /obj/item/device/robotanalyzer)))
+		I.attack(occupant, user)
+		return
+
+	if(default_deconstruction_screwdriver(user, I))
+		return
+	else if(default_deconstruction_crowbar(user, I))
+		if(occupant && avatar)
+			avatar.exit_vr()
+			avatar = null
+			go_out()
+		return
+
+
+/obj/machinery/vr_sleeper/MouseDrop_T(var/mob/target, var/mob/user)
+	if(user.stat || user.lying || !Adjacent(user) || !target.Adjacent(user)|| !isliving(target))
+		return
+	go_in(target, user)
+
+
+
+/obj/machinery/sleeper/relaymove(var/mob/user)
+	..()
+	if(usr.incapacitated())
+		return
+	go_out()
+
+
+
+/obj/machinery/vr_sleeper/emp_act(var/severity)
+	if(stat & (BROKEN|NOPOWER))
+		..(severity)
+		return
+
+	if(occupant)
+		// This will eject the user from VR
+		// ### Fry the brain? Yes. Maybe.
+		if(prob(severity * 15) && occupant.species.has_organ[O_BRAIN] && occupant.internal_organs_by_name[O_BRAIN])
+			var/obj/item/organ/O = occupant.internal_organs_by_name[O_BRAIN]
+			O.take_damage(severity * 2)
+			visible_message("<span class='danger'>\The [src]'s internal lighting flashes rapidly, before the hatch swings open with a cloud of smoke.</span>")
+			smoke.set_up(n = severity, 0, src)
+			smoke.start("#202020")
+		go_out()
+
+	..(severity)
+
+/obj/machinery/vr_sleeper/verb/eject()
+	set src in view(1)
+	set category = "Object"
+	set name = "Eject VR Capsule"
+
+	if(usr.incapacitated())
+		return
+
+	var/forced
```

Comment:
Is this a YES/NO var? If so, further references should be TRUE/FALSE, and this might be better if it's set to FALSE by default.

Reply:
The only time it isn't forced is when the eject verb is used, so defaulting to TRUE means it's less things to type, and I like most people want to type as little as possible.